src/main/java/org/testng/TestNG.java

```diff
@@ -290,8 +290,12 @@ private void parseSuite(String suitePath) {
       Collection<XmlSuite> allSuites = Parser.parse(suitePath, getProcessor());
 
       for (XmlSuite s : allSuites) {
-        s.setParallel(this.m_parallelMode);
-        s.setThreadCount(this.m_threadCount);
+        if (this.m_parallelMode != null) {
```

Comment:
`m_parallelMode` and `m_threadCount` are supposed to be valid values (default values if not set).
I'd prefer we sanitize the values when they are set. Do you think it is possible?

Reply:
There are two scenarios. 
1. When the configuration is set via the `TestNG` object (which am guessing is the CLI mode) 
2. When the configuration is set via the `XmlSuite` object (which am guessing is the xml mode)

The problem is that, when the user is on the xml mode, we end up overwriting values from the CLI mode. The easiest way of not doing it was to basically ensure that the CLI values dont override the xml values, when the CLI values aren't set. Sanitizing the values to some defaults would basically not help us in doing that, because we wouldn't know if it was the default values or if the user did set them explicitly to the default values. I chose the easier way of overriding values if they weren't set (`null` and `-1` respectively for `m_parallelMode` and `m_threadCount`)

Reply:
Ok, it makes sense.